src/Init/Lean/Elab/App.lean

```diff
@@ -180,7 +180,7 @@ unless (ctx.explicit || ctx.foundExplicit || ctx.typeMVars.isEmpty)  $ do
     | some eTypeBody =>
       unless eTypeBody.hasLooseBVars $
       when (hasTypeMVar ctx eTypeBody && hasOnlyTypeMVar ctx eTypeBody) $ do
-        isDefEq ctx.ref expectedType eTypeBody;
+        _ ← isDefEq ctx.ref expectedType eTypeBody;
```

Comment:
I really like that these changes make it clear that we don't run `isDefEq` here for an exception, but actively ignore the result and keep going.